ohttp/src/nss/aead.rs

```diff
@@ -149,6 +149,9 @@ impl Aead {
 
     pub fn open(&mut self, aad: &[u8], seq: SequenceNumber, ct: &[u8]) -> Res<Vec<u8>> {
         assert_eq!(self.mode, Mode::Decrypt);
+        if ct.len() < TAG_LEN {
```

Comment:
You could also do:

```rust
let pt_expected = ct.len().checked_sub(TAG_LEN).ok_or(Error::Truncated)?;
```
with appropriate use of this value below.

Reply:
Oh I like that much better - thanks!